### ISSUE TITLE
materialize is not loaded by plugin

### ISSUE THREAD

**Issue by manuel-guilbault** [manuel-guilbault/aurelia-materialize#2]

**Comment by tlvenn** [comment]
Was about to ask if there was any reason why there is no dependency on materialize-css :)

**Comment by manuel-guilbault** [comment]
Just didn't have time to do it :)

**Comment by manuel-guilbault** [comment]
@Tlvenn if you know how to work with shims and systemjs, I could use a hand: I can't figure out how to make it work

**Comment by tlvenn** [comment]
Bonjour Manuel,

It seems someone tried already to load Materialize with jspm and Guy Bedford helped figure out how to do this properly:
https://github.com/jspm/registry/issues/224

That being said, from what I can tell, Materialize has more dependencies than just Jquery. Reading https://github.com/Dogfalo/materialize/issues/634  it seems that there are 3 dependencies in total:
- JQuery
- HammerJS
- Velocity

Using the pointers from the first link, I hope it should now be easy to shim properly Materialize.

**Comment by tlvenn** [comment]
Given that Materialize is in the jspm registry already, I am not sure much is needed anymore...

http://kasperlewau.github.io/registry/#/?q=materialize

**Comment by tlvenn** [comment]
Or not given it only declare Jquery as dependency...
https://github.com/jspm/registry/blob/master/package-overrides/github/Dogfalo/materialize%400.96.1.json

Probably need to create a PR to update this.

**Comment by manuel-guilbault** [comment]
It now works as a JSPM dependency. The only thing I don't really like is that the `Materialize` object is still put on the global scope, and is not exported. I'll take some time to check the links you provided, and see if I can fix this.

Thanks for your help, and if you wish to contribute more, PRs are welcome :)

**Comment by tlvenn** [comment]
Will do, maybe we could create an issue with a todo list of the materialize components that need attention to leverage aurelia bindings. Might be a good starting point. Kind of a roadmap.

Getting back to dependencies, i checked and velocty/hammerJs are actually inlined in materialize js which left us with Jquery alone as external dependency. If you are using either of those 2 libs, you will end up with twice the code for those...

**Comment by manuel-guilbault** [comment]
The roadmap is a good idea; I'll create a wiki page for this.

As for materialize's dependencies, it's not ideal, but there's not much we can do about it. We can either deal with it or stop using the javascript part of materialize and implement all the widgets ourselves from scratch. I don't think that second option is worth the effort for now. Let's deal with this issue later.

**Comment by tlvenn** [comment]
Yes, trying to see if we can get Materialize to push a minimal lib with no dependencies at all. Then with Jspm override, we can declare those dependencies and point to that file as the main file.

**Comment by manuel-guilbault** [comment]
Great!

**Comment by tlvenn** [comment]
For reference: https://github.com/Dogfalo/materialize/issues/1741
